miasm2/arch/aarch64/sem.py

```diff
@@ -730,6 +793,36 @@ def irbloc_fix_regs_for_mode(self, irbloc, mode=64):
                 irs[i] = self.expr_fix_regs_for_mode(e)
         irbloc.dst = self.expr_fix_regs_for_mode(irbloc.dst)
 
+    def mod_pc(self, instr, instr_ir, extra_ir):
+        "Replace PC by the instruction's offset"
+        cur_offset = m2_expr.ExprInt64(instr.offset)
+        for i, expr in enumerate(instr_ir):
+            dst, src = expr.dst, expr.src
+            if dst != self.pc:
+                dst = dst.replace_expr({self.pc: cur_offset})
+            src = src.replace_expr({self.pc: cur_offset})
+            instr_ir[i] = m2_expr.ExprAff(dst, src)
+        for b in extra_ir:
+            for irs in b.irs:
+                for i, expr in enumerate(irs):
+                    dst, src = expr.dst, expr.src
+                    if dst != self.pc:
+                        dst = dst.replace_expr({self.pc: cur_offset})
+                    src = src.replace_expr({self.pc: cur_offset})
+                    irs[i] = m2_expr.ExprAff(dst, src)
+
+
+    def del_dst_zr(self, instr, instr_ir, extra_ir):
+        "Writes to zero register are discarded"
+        instr_ir = [expr for expr in instr_ir if expr.dst not in [WZR, XZR]]
+
+        for b in extra_ir:
+            for i, irs in eunmerate(b.irs):
+                irs = [expr for expr in irs if expr.dst not in [WZR, XZR]]
```

Comment:
Here, you rewrite `irs`, the loop variable. You may merge this line with the next one:
`b.irs[i] = [expr for expr in irs if expr.dst not in [WZR, XZR]]`
Or with a `map`:
`b = map(lambda irs:  [expr for expr in irs if expr.dst not in [WZR, XZR]], b)`

In addition, `[WZR, XZR]` is redefined in each iteration, you should get it off the loop inside a `CONST`.